lib/web/apiserver.go

```diff
@@ -2865,6 +2868,43 @@ func (h *Handler) getClusters(w http.ResponseWriter, r *http.Request, p httprout
 	return out, nil
 }
 
+type GetClusterInfoResponse struct {
+	ui.Cluster
+	IsCloud bool `json:"isCloud"`
+}
+
+// getClusterInfo returns the information about the cluster in the :site param
+//
+// GET /v1/webapi/sites/:site/info
+//
+// Successful response:
+//
+//	{"name": "localhost","lastConnected": "RFC3339 time","status": "online","publicURL": "localhost:3080","authVersion": "17.0.0-dev","proxyVersion": "17.0.0-dev", "isCloud": true}
+func (h *Handler) getClusterInfo(w http.ResponseWriter, r *http.Request, p httprouter.Params, sctx *SessionContext, site reversetunnelclient.RemoteSite) (interface{}, error) {
+	ctx := r.Context()
+	c, err := ui.GetClusterDetails(ctx, site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	clt, err := sctx.GetUserClient(ctx, site)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	pingResp, err := clt.Ping(ctx)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	isCloud := pingResp.GetServerFeatures().Cloud
+
+	return GetClusterInfoResponse{
+		Cluster: *c,
+		IsCloud: isCloud,
+	}, nil
```

Comment:
nit: not a hard suggestion and up to you if you want to keep it the way it is, but since `pingResp.GetServerFeatures().Cloud` is reasonable short, we can directly use it. 
```suggestion

	return GetClusterInfoResponse{
		Cluster: *c,
		IsCloud: pingResp.GetServerFeatures().Cloud,
	}, nil
```